### PR TITLE
Fixed: upload button gets enabled before queue is empty, closes #432

### LINKED ISSUE #432: Upload button gets enabled too early when queue is submitted
**Describe the bug**
In the offline capable view, with a queue of multiple records, the upload button is shown as busy when uploads start but becomes not-busy as soon as the first record is submitted. There may be a more serious bug as well, where records could theoretically be submitted multiple times, if the queue takes more than 5 minutes to get submitted.

**To Reproduce**
Easiest is to use a form with file upload and slow to create large records, e.g.

```curl
curl --user enketorules: -d "server_url=https://enketo-aggregate.appspot.com&form_id=files-multiple" http://localhost:8005/api/v2/survey/offline
```

1. load form and go offline
2. at 3 (or more) large finalized records to the queue
3. go online and click Upload
4. See the button busy state changing to not busy after the first record is submitted.

**Expected behavior**
For the Upload button to stay busy until the last record is submitted.

### PR DESCRIPTION
**Pull request by MartijnR**
Closes #432

#### I have verified this PR works with

-   [ ] Online form submission
-   [x] Offline form submission
-   [x] Saving offline drafts
-   [ ] Loading offline drafts
-   [ ] Editing submissions
-   [ ] Form preview
-   [ ] None of the above

### CODE REVIEW

File: public/js/src/module/records-queue.js [hunk 1]
```diff
@@ -288,38 +287,31 @@ function uploadQueue(byUser = false) {
...
                                     'error',
                                     errorMsg
                                 );
                             })
-                            .then(() => {
-                                $uploadButton.btnBusyState(false);
-                                if (
-                                    successes.length + fails.length ===
-                                    records.length
-                                ) {
```
Comment (MartijnR) [inline public/js/src/module/records-queue.js:303]: I do not know what I was thinking when I wrote this code long ago. I'm hoping that way back this made sense, but think it's better I don't find out.

This whole block should just move to after all submissions are completed/failed.

**Comment by MartijnR** [comment]
@eyelidness, just pinging to see if you have a chance to look at this. No hurry, but otherwise, OC will just merge into their fork first.

**Comment by MartijnR** [comment]
Thank you!

**Review by eyelidlessness — APPROVED** [review]
Thanks @MartijnR!

**Pull request merged**